### ISSUE TITLE
Most sensor entities become "not available" in HA after user logs out and log back in again UNLESS....

### ISSUE THREAD

**Issue by mkanet** [home-assistant/android#2724]
I currently use Nabu Casa remote login.  Every time any user logs out of their Android mobile app, then logs back in again, **_most_** of their mobile app's sensor entities will become "not available" in Home Assistant... even if they wait indefinitely or select for all sensors to update every minute.
![not-available](https://user-images.githubusercontent.com/811008/181670074-6420cdba-3fa1-4c10-aaac-76d8dde052c4.png)

The only way to make the affected "not available" sensor entities to display again in Home Assistant is going back into the respective mobile app, then select **Manage Sensors**; and toggle each affected sensor: **disable/enable**.  After doing this, each sensor entity will become available again immediately in Home Assistant.

**Note 1:** Disabling/enabling each affected sensor from the web interface (under Entities page) **will not** fix this issue.  It HAS to be done in the mobile apps themselves.

**Note 2:** This issue is also reproducible if I disable Nabu Casa remote login feature and access my Home Assistant instance directly via DDNS. 

**Note 3:** If logged in users never log out of their mobile app, the entities will continue to work indefinitely.  However, this isn't really feasible.

How can I avoid this issue from happening after logging out and logging back into each mobile app?  Or at least an easy way to toggle all existing enabled sensors to off and on again?  Maybe the app could be enhanced to reset all existing Enabled sensors after logging back in to ensure they will be available again in Home Assistant?  This minor enhancement wouldn't cause any issues, it would only help make the sensors more resilient when logging back in.

**Comment by dshokouhi** [comment]
Looks like the issue template is missing here so we are not seeing all requested information. If possible please copy it from a new issue and fill it in here so we can get all the important data. In this case we need to see the logs when the issue occurs so we can see what re-registration looks like.

Would also be good to confirm things like if the device name was changed etc.. after logging back in.

**Comment by mkanet** [comment]
My apologies.  Please find below information using a standard template:

**Describe the bug**
I currently use Nabu Casa remote login. Every time any user logs out of their Android mobile app, then logs back in again, many of their mobile app's sensor entities completely missing Home Assistant... even if they wait indefinitely or select for all sensors to update every minute.  

NOTE: Some of the missing entities DO get added to Home Assistant in the next 15 minutes.  However, many of them will never get added to Home Assistant regardless of how long I wait.
![181670074-6420cdba-3fa1-4c10-aaac-76d8dde052c4](https://user-images.githubusercontent.com/811008/181806334-8f3d8255-d764-492a-bda6-5884fc3cb589.png)

**To Reproduce**
1. Log out of Android Home Assistant Companion
2. Restart Android Home Assistant Companion
3. Log back into Android Home Assistant Companion

**Expected behavior**
After logging out and logging back in Home Assistant Companion, all respective enabled sensor entities in Home Assistant should be displayed within 15 minutes and updated just like before.

**System details**
 - Home-assistant (version):   2022.7.7
 - Hassio (Yes/No):  Yes  (VirtualBox VM on Windows 10 host)
 - Home Assistant Companion (version):  2022.6.0-full

**Logs**
Unsure what logger settings to use in order to capture/debug issue.  Default logger settings doesn't expose any information related to this issue.

**Additional context**
Device names did not changed at anytime (automatically or manually).

Things I've tried:
- Changing update frequency to every minute (and restarting mobile app afterwards)
- Restarting app (even tried deleting temporary cache before starting app again)
- Tried logging into my Home Assistant instance directly via DDNS instead of Nabu Casa remote login URL.
- Correction:  Disabling/re-enabling entities from the Home Assistant web interface (under Entities page) is NOT possible since these entities are **completely missing**.  They aren't displayed even as "disabled".
 
**Workaround**
Open mobile app, select "Manage Sensors", then manually disable/enable each sensor that's not being displayed in Home Assistant.  As soon as I disable/enable a sensor, it's respective entity is immediately displayed in Home Assistant and start polling again.  Needless to say, this is VERY time consuming!

**Suggestion**
In case there isn't a fix, it would be great if the mobile app offered the ability to quietly disable/enable all existing enabled sensors (either automatically after logging in or a manual optional setting for users).

**Comment by dshokouhi** [comment]
We are missing on device logs. Please go to Configuration > Companion App > Show and share logs then get us the logs after you reproduce the issue. We need to see the full set of logs here.

Also the issue template you copied is from the wrong repo, it needed to be from the android repo (where you created the issue)

**Comment by mkanet** [comment]
Please find the full set of logs below after reproducing the issue.  Please let me know if there's anything else I can provide.
[homeassistant_companion_log_6-29-2022_11-24-24.txt](https://github.com/home-assistant/android/files/9222471/homeassistant_companion_log_6-29-2022_11-24-24.txt)

**Comment by dshokouhi** [comment]
@jpelgrom what should happen in this case? It seems the sensors are not being checked against the core registration here. I wonder if maybe we need to clear out the registration in the DB so its forced to re-register next login?

**Comment by jpelgrom** [comment]
Sensors are always checked against the core registration, but I guess it isn't finding any difference in enabled state? Hard to know for sure without debug logs for the sensor requests (because that will include the server response).

> I wonder if maybe we need to clear out the registration in the DB so its forced to re-register next login?

I always assumed there was a reason the app isn't doing any cleanup of server-specific data like sensors, widgets, tiles config, ... on logout? This isn't a new bug caused by syncing disabled entities to core 2022.6+ as I understand it right now.

**Comment by dshokouhi** [comment]
> Sensors are always checked against the core registration, but I guess it isn't finding any difference in enabled state? Hard to know for sure without debug logs for the sensor requests (because that will include the server response).

yea its tough to tell here I agree but we can at least see the sensors attempting to update :)



> I always assumed there was a reason the app isn't doing any cleanup of server-specific data like sensors, widgets, tiles config, ... on logout?

I think that code was there since before we had sensor management and widgets etc...



> This isn't a new bug caused by syncing disabled entities to core 2022.6+ as I understand it right now.

Correct not new, this was pre-existing. Logging out and back in is not something I expect to see in typical use cases. More of an edge case for our app which is probably why we are just hearing about it :)

**Comment by jpelgrom** [comment]
FYI managed to reproduce this, the server response for `entities` from `get_config` which is used to sync entity state only includes one entity:

```json
{
   "latitude":52.01666538599529,
   "longitude":4.35330722973883,
   "elevation":4,
   "unit_system":{
      "length":"km",
      "accumulated_precipitation":"mm",
      "mass":"g",
      "pressure":"Pa",
      "temperature":"\u00b0C",
      "volume":"L",
      "wind_speed":"m/s"
   },
   "location_name":"Home",
   "time_zone":"Europe/Amsterdam",
   "components":[ ...   ],
   "version":"2022.7.7",
   "theme_color":"#03A9F4",
   "entities":{
      "bd78a79654bb7d61":{
         "disabled":false
      }
   }
}
```

The app's registration check is very safe and doesn't distinguish between no response or a response but the key wasn't included. Because it still has a registered state in the database (from before logout) it will not send a new registration request for the sensor. I could adjust it to re-register it in this case as well, will do some additional testing and submit it later.

> Logging out and back in is not something I expect to see in typical use cases. More of an edge case for our app which is probably why we are just hearing about it :)

Definitely an edge case. Clearing all server-related data on logout is something we should look at, might fix a lot of the 'please clear all the app data' issues.

**Comment by jpelgrom** [comment]
FYI managed to reproduce this, the server response for `entities` from `get_config` which is used to sync entity state only includes one entity:

```json
{
   "latitude": 0,
   "longitude": 0,
   "elevation": 0,
   "unit_system":{
      "length":"km",
      "accumulated_precipitation":"mm",
      "mass":"g",
      "pressure":"Pa",
      "temperature":"\u00b0C",
      "volume":"L",
      "wind_speed":"m/s"
   },
   "location_name":"Home",
   "time_zone":"Europe/Amsterdam",
   "components":[ ...   ],
   "version":"2022.7.7",
   "theme_color":"#03A9F4",
   "entities":{
      "bd78a79654bb7d61":{
         "disabled":false
      }
   }
}
```

The app's registration check is very safe and doesn't distinguish between no response or a response but the key wasn't included. Because it still has a registered state in the database (from before logout) it will not send a new registration request for the sensor. I could adjust it to re-register it in this case as well, will do some additional testing and submit it later.

> Logging out and back in is not something I expect to see in typical use cases. More of an edge case for our app which is probably why we are just hearing about it :)

Definitely an edge case. Clearing all server-related data on logout is something we should look at, might fix a lot of the 'please clear all the app data' issues.

**Comment by mkanet** [comment]
Guys, thanks so much for looking into this; and, being able to reproduce this.

FYI, I vaguely remember noticing this issue a while back before HA Version 2022.6. I had to disable/enable the affected sensors.  However, I just shrugged off this issue since our family members rarely logout and log back in the mobile apps.

Thanks again.
